### PR TITLE
chore: Document the usage of the che-server API

### PR DESCRIPTION
**Pull request by tolusha**
Signed-off-by: Anatolii Bazko <abazko@redhat.com>

### What does this PR do?
Document the usage of the che-server API

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/21848

### Is it tested? How?

N/A

#### Release Notes

N/A

#### Docs PR

N/A

### CODE REVIEW

**Review by ibuziuk — COMMENTED** [review]
@akurinnoy @olexii4 folks, could you please review and confirm that those are the only API used by the dashboard

File: README.md [hunk 1]
```diff
@@ -226,6 +226,23 @@ kubectl rollout restart deployment/che-operator -n $CHE_NAMESPACE
...
+| Method | Path                            |
+|--------|---------------------------------|
+| GET    | /kubernetes/namespace/provision |
+| POST   | /kubernetes/namespace           |
+| POST   | /factory/resolver/              |
+| POST   | /factory/token/refresh          |
+| GET    | /workspace/settings             |
+| GET    | /oauth                          |
+| GET    | /oauth/authenticate             |
+| GET    | /oauth/callback                 |
```
Comment (olexii4) [inline README.md:242]: BTW: [che-workspace-client](https://github.com/eclipse-che/che-workspace-client) doesn't support  bouth ```/oauth/callback``` and ```/oauth/authenticate```.
I think we should remove them.
@akurinnoy WDYT?

Comment (tolusha) [reply README.md:242]: I saw both `/oauth/callback` and `/oauth/authenticate` in network tab. Where do they come from?

Comment (olexii4) [reply README.md:242]: @tolusha Oh, I forgot about it. We need them for the authorization flow. We need them.

Comment (akurinnoy) [reply README.md]: @olexii4 those requests are not initiated by the dashboard

**Review by akurinnoy — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (ibuziuk) [reply README.md]: @olexii4 @akurinnoy folks, but who is doing `/oauth/callback` && `/oauth/authenticate` ?

**Pull request merged**